keyboards/annepro2/boards/ANNEPRO2_C15/board.c

```diff
@@ -0,0 +1,119 @@
+
+/**
+ * @brief   PAL setup.
+ * @details Digital I/O ports static configuration as defined in @p board.h.
+ *          This variable is used by the HAL when initializing the PAL driver.
+ */
+const PALConfig pal_default_config = {
+    // GPIO A
+    .setup[0] =
+        {
+            .DIR    = OUT_BITS(IOPORTA),
+            .INE    = IN_BITS(IOPORTA),
+            .PU     = IN_BITS(IOPORTA),
+            .PD     = 0x0000,
+            .OD     = 0x0000,
+            .DRV    = 0x0000,
+            .LOCK   = 0x0000,
+            .OUT    = 0x0000,
+            .CFG[0] = AF_BITS(IOPORTA, 0),
+            .CFG[1] = AF_BITS(IOPORTA, 1),
+        },
+    // GPIO B
+    .setup[1] =
+        {
+            .DIR    = OUT_BITS(IOPORTB),
+            .INE    = IN_BITS(IOPORTB),
+            .PU     = IN_BITS(IOPORTB),
+            .PD     = 0x0000,
+            .OD     = 0x0000,
+            .DRV    = 0x0000,
+            .LOCK   = 0x0000,
+            .OUT    = 0x0000,
+            .CFG[0] = AF_BITS(IOPORTB, 0),
+            .CFG[1] = AF_BITS(IOPORTB, 1),
+        },
+    // GPIO C
+    .setup[2] =
+        {
+            .DIR    = OUT_BITS(IOPORTC),
+            .INE    = IN_BITS(IOPORTC),
+            .PU     = IN_BITS(IOPORTC),
+            .PD     = 0x0000,
+            .OD     = 0x0000,
+            .DRV    = 0x0000,
+            .LOCK   = 0x0000,
+            .OUT    = 0x0000,
+            .CFG[0] = AF_BITS(IOPORTC, 0),
+            .CFG[1] = AF_BITS(IOPORTC, 1),
+        },
+    // GPIO D
+    .setup[3] =
+        {
+            .DIR    = OUT_BITS(IOPORTD),
+            .INE    = IN_BITS(IOPORTD),
+            .PU     = IN_BITS(IOPORTD),
+            .PD     = 0x0000,
+            .OD     = 0x0000,
+            .DRV    = 0x0000,
+            .LOCK   = 0x0000,
+            .OUT    = 0x0000,
+            .CFG[0] = AF_BITS(IOPORTD, 0),
+            .CFG[1] = AF_BITS(IOPORTD, 1),
+        },
+    .ESSR[0] = 0x00000000,
+    .ESSR[1] = 0x00000000,
+};
+
+const ioline_t row_list[MATRIX_ROWS] = {
+    LINE_ROW1, LINE_ROW2, LINE_ROW3, LINE_ROW4, LINE_ROW5,
+};
+
+const ioline_t col_list[MATRIX_COLS] = {
+    LINE_COL1, LINE_COL2, LINE_COL3, LINE_COL4, LINE_COL5, LINE_COL6, LINE_COL7, LINE_COL8, LINE_COL9, LINE_COL10, LINE_COL11, LINE_COL12, LINE_COL13, LINE_COL14,
+};
+
+void __early_init(void) {}
+
+void early_hardware_init_pre(void) { ht32_clock_init(); }
+
+void board_init(void) {}
+
+void boardInit(void) {}
```

Comment:
These are QMK-specific constructs and do not belong in the `board.c`.
You've misinterpreted the intent here -- `ht32_clock_init();` does indeed go into `__early_init()`, as per normal board files for stm32, which place `stm32_clock_init()` in the same location.

As I pointed out on specific lines highlighted in the previous comment, the issue was the `gpt_init()` and the `spi_init()`, not the placement of the clock init.

Refer to an existing ChibiOS board file for example structure.

```suggestion
void __early_init(void) { ht32_clock_init(); }

void boardInit(void) {}
```